optimum/onnx/graph_transformations.py

```diff
@@ -13,28 +13,39 @@
 #  limitations under the License.
 import copy
 from collections import defaultdict
-from typing import DefaultDict, Dict, Set, Tuple
+from typing import DefaultDict, Dict, List, Set, Tuple
 
-from onnx import ModelProto
+import onnx
+from onnx import ModelProto, ValueInfoProto
 
 
 def _find_duplicate_weights(model) -> DefaultDict[Tuple[int, bytes], Set[str]]:
+    return _find_duplicate_initializers(model.graph.initializer)
+
+
+def _find_duplicate_initializers(initializers) -> DefaultDict[Tuple[int, bytes], Set[str]]:
     duplicates = defaultdict(set)
-    for initializer in model.graph.initializer:
+    for initializer in initializers:
+        tensor_dims = tuple(getattr(initializer, "dims"))
         for data_attr in ["raw_data", "int32_data", "int64_data", "uint64_data", "float_data", "double_data"]:
             tensor_data = getattr(initializer, data_attr)
             if tensor_data:
                 tensor_data = tuple(tensor_data)
                 break
-        duplicates[(initializer.data_type, tensor_data)].add(initializer.name)
+        duplicates[(initializer.data_type, tensor_data, tensor_dims)].add(initializer.name)
```

Comment:
Here I addded tensor_dims, I doubt if I met some small initializers(constant) with the same tensor_data but different data_type / tensor_dims.

Reply:
Maybe it's best to keep as a tuple then until we validate further. The `to_array` + hash trick seem to work fine both memory-wise and time-wise.